Xamarin.Forms.Core/RadioButton.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using Xamarin.Forms.Platform;
+
+namespace Xamarin.Forms
+{
+	[RenderWith(typeof(_RadioButtonRenderer))]
+	public partial class RadioButton : Button
+	{
+		static Dictionary<string, List<WeakReference<RadioButton>>> _groupNameToElements;
+
+		public const string IsCheckedVisualState = "IsChecked";
+
+		public static readonly BindableProperty IsCheckedProperty = BindableProperty.Create(
+			nameof(IsChecked), typeof(bool), typeof(RadioButton), false, propertyChanged: (b, o, n) => ((RadioButton)b).OnIsCheckedPropertyChanged((bool)n));
+
+		public static readonly BindableProperty GroupNameProperty = BindableProperty.Create(
+			nameof(GroupName), typeof(string), typeof(RadioButton), null, propertyChanged: (b, o, n) => ((RadioButton)b).OnGroupNamePropertyChanged((string)o, (string)n));
+
+		public static readonly BindableProperty ButtonSourceProperty = BindableProperty.Create(
+			nameof(ButtonSource), typeof(ImageSource), typeof(RadioButton), null);
+
+		public event EventHandler<CheckedChangedEventArgs> CheckedChanged;
+
+		public bool IsChecked
+		{
+			get { return (bool)GetValue(IsCheckedProperty); }
+			set { SetValue(IsCheckedProperty, value); }
+		}
+
+		public string GroupName
```

Comment:
It looks like the scope of the group name is global, meaning you can't have 2 instances of the same custom control alive at once.

How is this done in other platforms ?

Reply:
I don't understand what you mean by it's "global". You definitely can have two instances with the same group name, but it depends if the two have the same parent or not. You mentioned something similar in a previous comment, and I replied to that too. I explained the group behavior in the description, there's also a gallery page showing the  group name behavior in different scenarios, and I made a video with the page gallery.

Reply:
ok, so the scope is name + parent ?

Reply:
so what if the individual radio button are nested within frames ?

Reply:
> ok, so the scope is name + parent ?

yes

> so what if the individual radio button are nested within frames ?

what about it?

Reply:
you have
```
StackLayout
  Frame
    RadioButton
  Frame
    RadioButton
```

so the 2 radio buttons aren't in the same group ?

Reply:
also, you say the scope is the parent, but looking at GetVisualRoot, the scope is the page. what if I have this

```
Page
  StackLayout
    YesNoMaybeControl
    YesNoMaybeControl
    YesNoMaybeControl
```